pkg/registry/secrets.go

```diff
@@ -0,0 +1,245 @@
+package registry
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	clusterv1beta1 "github.com/replicatedhq/embedded-cluster-kinds/apis/v1beta1"
+	ectypes "github.com/replicatedhq/embedded-cluster-kinds/types"
+	"github.com/replicatedhq/embedded-cluster-operator/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const (
+	// SeaweedfsS3SecretReadyConditionType represents the condition type that indicates status of
+	// the Seaweedfs secret.
+	SeaweedfsS3SecretReadyConditionType = "SeaweedfsS3SecretReady"
+
+	// RegistryS3SecretReadyConditionType represents the condition type that indicates status of
+	// the Registry secret.
+	RegistryS3SecretReadyConditionType = "RegistryS3SecretReady"
+)
+
+func EnsureSecrets(ctx context.Context, in *clusterv1beta1.Installation, metadata *ectypes.ReleaseMetadata, cli client.Client) error {
+	log := ctrl.LoggerFrom(ctx)
+
+	config, op, err := ensureSeaweedfsS3Secret(ctx, in, metadata, cli)
+	if err != nil {
+		in.Status.SetCondition(metav1.Condition{
+			Type:               SeaweedfsS3SecretReadyConditionType,
+			Status:             metav1.ConditionFalse,
```

Comment:
I think you need to check whether this is HA before you create these secrets
Maybe set a condition of "secret not needed" or something?